deepwell/src/config/special_action.rs

```diff
@@ -0,0 +1,72 @@
+/*
+ * config/special_action.rs
+ *
+ * DEEPWELL - Wikijump API provider and database manager
+ * Copyright (C) 2019-2023 Wikijump Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//! Perform a "special action" instead of normal server execution.
+//!
+//! This is useful in contexts such as CI, where we want DEEPWELL to
+//! not run as a daemon, but instead perform a special action or check,
+//! as if motivated by a script.
+
+use super::Config;
+use std::path::PathBuf;
+use std::{env, process};
+
+pub fn run_special_action() {
+    // Get action name, if specified.
+    // Otherwise return and perform normal execution.
+    let action_name = match env::var("DEEPWELL_SPECIAL_ACTION") {
+        Ok(value) => value,
+        Err(_) => return,
+    };
```

Comment:
I believe patterns like this could be simplified into just this one-liner:
```suggestion
    let Ok(action_name) = env::var("DEEPWELL_SPECIAL_ACTION") else { return };
```